scripts/fix_lotus_encumbrances/fix_lotus_encumbrances.py

```diff
@@ -496,22 +511,97 @@ async def release_unreleased_encumbrances_for_closed_orders(closed_orders_ids, f
     print('  Released {} encumbrances.'.format(sum(nb_released_encumbrances)))
 
 
-def get_fy_fund_ids(fiscal_year_id):
-    query = 'fyscal_year.id=={}'.format(fiscal_year_id)
-
-    params = {'query': query, 'offset': '0', 'limit': ITEM_MAX}
+def get_funds():
+    params = {'offset': '0', 'limit': ITEM_MAX}
     try:
-        r = requests.get(okapi_url + 'finance/funds', headers=headers, params=params)
+        r = requests.get(okapi_url + 'finance-storage/funds', headers=headers, params=params)
         if r.status_code != 200:
             raise_exception_for_reply(r)
         funds = r.json()['funds']
-        ids = []
-        for fund in funds:
-            ids.append(fund.get('id'))
     except Exception as err:
         print('Error getting order ids with query "{}": {}'.format(query, err))
         raise SystemExit(1)
-    return ids
+    return funds
+
+
+async def get_polines_by_order_id(order_id):
+    query = 'purchaseOrderId=={}'.format(order_id)
+    resp = await get_request(okapi_url + 'orders-storage/po-lines', query)
+    po_lines = resp['poLines']
+    return po_lines
+
+
+def lookup_fund(poline_id, code, all_funds):
+    for fund in all_funds:
+        if fund['code'] == code:
+            return fund['id']
+    print('Fund code {} from poline {} not exists. Correct the data and rerun the script'.format(code, poline_id))
+    raise SystemExit(1)
+
+
+async def process_order_encumbrances_relations(order_id, fiscal_year_id, all_funds, order_sem):
+    po_lines = await get_polines_by_order_id(order_id)
+
+    poline_futures = []
+    if len(po_lines) > 0:
+        for po_line in po_lines:
+            poline_futures.append(asyncio.ensure_future(process_po_line_encumbrances_relations(po_line, fiscal_year_id, all_funds)))
+        await asyncio.gather(*poline_futures)
+
+    order_sem.release()
+
+
+async def check_if_fd_needs_updates(poline_id, fd, query):
+    encumbrances = await get_encumbrances_by_query(query)
+    if len(encumbrances) > 0 and encumbrances[0]['id'] != fd['encumbrance']:
+        print('Updating poline {} encumbrance {} with new value {}'.format(poline_id, fd['encumbrance'], encumbrances[0]['id']))
+        fd['encumbrance'] = encumbrances[0]['id']
+        return True
+    return False
+
+
+async def process_po_line_encumbrances_relations(poline, fiscal_year_id, all_funds):
+    enc_futures = []
+    for fd in poline['fundDistribution']:
+        if fd.get('encumbrance') is not None:
+            query = 'amount<>0.0 AND encumbrance.sourcePoLineId=={} AND fiscalYearId<>{} AND fromFundId=={}'.format(
```

Comment:
I believe query is incorrect
1. `encumbrance.sourcePoLineId=={}` should be line id, but `poline['purchaseOrderId']` is used, so it should be `poline['id']`
2. `fiscalYearId<>{}` should be `fiscalYearId=={}` as you try to find unlinked encumbrances in current FY, but <> means that in previous

in addition we should add a note, that this script can be executed only for current year, as we'll define links for previous years. Or we should check that FY is current one and skip step to fix references

Reply:
fixed